hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.monster.dap
+
+import org.broadinstitute.monster.dogaging.jadeschema.table.HlesHealthCondition
+
+object HealthTransformations {
+
+  /** Parse all health-condition-related fields out of a raw RedCap record. */
+  def mapHealthConditions(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    // condition type is congenital vs infectious vs tox vs trauma vs etc
+    health_conditions.filter { case (condition, _) => rawRecord.getBoolean(s"hs_${condition}_yn") }.flatMap {
+      // infectious disease
+      case ("dx_infectious", conditionCategorical) =>
+        // process infectious disease data
+        infectious_diseases.filter {
+          case (infection, _) => rawRecord.getBoolean(s"hs_dx_${infection}")
+        }.map {
+          case (infection, infectionCategorical) =>
+            HlesHealthCondition(
+              dogId = rawRecord.getRequired("study_id").toLong,
+              hsConditionType = conditionCategorical.toLong,
+              hsCondition = infectionCategorical.toLong,
+              // should only be populated if hs_dx_infect_other is the current infectionCategorical (value is 40)
+              hsConditionOtherDescription = if (infectionCategorical == 40) {
+                rawRecord.getOptional("hs_dx_infect_other_spec")
+              } else { None },
+              // infectious diseases are not categorized as congenital
+              hsConditionIsCongenital = false,
+              // no relevant field for infectious diseases
+              hsConditionCause = None,
+              // no relevant field for infectious diseases
+              hsConditionCauseOtherDescription = None,
+              hsDiagnosisYear = rawRecord.getOptionalNumber(s"hs_dx_${infection}_year"),
+              hsDiagnosisMonth = rawRecord.getOptionalNumber(s"hs_dx_${infection}_month"),
+              hsRequiredSurgeryOrHospitalization =
+                rawRecord.getOptionalNumber(s"hs_dx_${infection}_surg"),
+              hsFollowUpOngoing = rawRecord.getOptionalBoolean(s"hs_dx_${infection}_fu")
+            )
+        }
+      // an "else" case to please the compiler gods for now, will eventually be replaced by all the other health
+      // condition cases
+      case (_, conditionCategorical) =>
+        List(
+          HlesHealthCondition(
+            dogId = rawRecord.getRequired("study_id").toLong,
+            hsConditionType = conditionCategorical.toLong,
+            hsCondition = 100L,
+            hsConditionOtherDescription = Some("falafel"),
+            hsConditionIsCongenital = false,
+            hsConditionCause = None,
+            hsConditionCauseOtherDescription = None,
+            hsDiagnosisYear = Some(2020),
+            hsDiagnosisMonth = Some(3),
+            hsRequiredSurgeryOrHospitalization = Some(1),
+            hsFollowUpOngoing = Some(true)
+          )
+        )
+    }
+
+  val health_conditions: List[(String, Int)] = List(
```

Comment:
for both these and for the infectious diseases, I am laying them out in the order they appear on RedCap in the data dictionary and then generating a categorical integer for them by their indexed position. I'm also open to explicitly writing out the categorical integer each one maps to; the only benefit I'd see there is that we, the developers, could clearly read the number. Otherwise, so long as we maintain the order here, it should operate the same.